ckanext/dbquery/blueprints/dbquery.py

```diff
@@ -18,17 +20,58 @@ def index():
     result = None
     request = toolkit.request
 
+    # Check if we're loading a query from history
+    load_query_id = request.args.get('load_query')
+    if load_query_id:
+        # Find the query by ID
+        saved_query = model.Session.query(DBQueryExecuted).filter(
+            DBQueryExecuted.id == load_query_id
+        ).first()
+
+        if saved_query:
+            # Pre-populate the query field
+            query = saved_query.query
```

Comment:
If we want a custom view for a single query, it must be a different view. It's ll be better
This is something from a diffrent PR, is not required in the base issue
We can drop it and do it in a new issue.